src/main/scala/shine/DPIA/Semantics/OperationalSemantics.scala

```diff
@@ -18,6 +18,9 @@ object OperationalSemantics {
   final case class IntData(i: Int) extends Data(int) {
     override def toString: String = i.toString
   }
+  final case class HalfData(f: Float) extends Data(f16) {
+    override def toString: String = "__float2half(" + f.toString + "f)"
```

Comment:
maybe generating `__float2half` should be specific to the CUDA code generation instead of a generic DPIA print.